packages/wonder-blocks-core/util/i18n.js

```diff
@@ -0,0 +1,266 @@
+// @flow
+
+const interpolationMarker = /%\(([\w_]+)\)s/g;
+
+/**
+ * Simple i18n method with sprintf-like %(name)s replacement
+ * To be used like so:
+ *   i18n._("Some string")
+ *   i18n._("Hello %(name)s", {name: "John"})
+ */
+const _ = (
+    str: string | {messages: Array<string>},
+    options?: {[key: string]: string | number},
+) => {
+    // Sometimes we're given an argument that's meant for ngettext().  This
+    // happens if the same string is used in both i18n._() and i18n.ngettext()
+    // (.g. a = i18n._(foo); b = i18n.ngettext("foo", "bar", count);
+    // In such cases, only the plural form ends up in the .po file, and
+    // then it gets sent to us for the i18n._() case too.  No problem, though:
+    // we'll just take the singular arg.
+    if (typeof str === "object" && str.messages) {
+        str = str.messages[0];
+    }
+
+    return str.replace(interpolationMarker, (match, key) => {
+        if (options && options.hasOwnProperty(key)) {
+            const replaceWith = options[key];
```

Comment:
If `replaceWith` is a `number` it would be nice to call `toLocaleString` on it, it's really only need when called from `ngettext`.  We'd also have to pass the `lang` to `_()` to be able to use it.